docs/v3/deploy/infrastructure-concepts/store-flow-code.mdx

```diff
@@ -1,32 +1,109 @@
+As detailed in the [Deployment overview](/v3/deploy), you can create a deployment in one of two main ways:
+
+- [Python code with the `flow.deploy` method](/v3/deploy/infrastructure-concepts/deploy-via-python)
+
+    - When using `.deploy`, specify a storage location for your flow with the `flow.from_source` method.
+
+    - The `source` is either a URL to a git repository or a storage object. For example:
+        - A local directory: `source=Path(__file__).parent` or `source="/path/to/file"`
+        - A URL to a git repository: `source="https://github.com/org/my-repo.git"`
+        - A storage object: `source=GitRepository(url="https://github.com/org/my-repo.git")`
+
+    - The `entrypoint` is the path to the file the flow is located in and the function name, separated by a colon.
+
+
+- [YAML specification defined in a `prefect.yaml` file](/v3/deploy/infrastructure-concepts/prefect-yaml)
+
+    - To create a `prefect.yaml` file interactively, run `prefect deploy` from the CLI and follow the prompts.
+
+    - The `prefect.yaml` file may define a `pull` section that specifies the storage location for your flow. For example:
+        - Set the working directory:
+        ```yaml
+        pull:
+            - prefect.deployments.steps.set_working_directory:
+                directory: /path/to/directory
+        ```
+        - Clone a git repository:
+        ```yaml
+        pull:
+            - prefect.deployments.steps.git_clone:
+                repository: https://github.com/org/my-repo.git
+        ```
+        - Pull from blob storage:
+        ```yaml
+        pull:
+            - prefect.deployments.steps.pull_from_blob_storage:
+                container: my-container
+                folder: my-folder
+        ```
+
+Whether you use `from_source` or `prefect.yaml` to specify the storage location for your flow code, the resulting deployment will have a set of `pull` steps that your worker will use to retrieve the flow code at runtime.
+
+
+## Store code locally
+
+If using a Process work pool, you can use one of the remote code storage options shown above, or you can store your flow code in a local folder.
+
+Here is an example of how to create a deployment with flow code stored locally:
+
+<CodeGroup>
+
+```python local_process_deploy_local_code.py
+from prefect import flow
+from pathlib import Path
+
+
+@flow(log_prints=True)
+def my_flow(name: str = "World"):
+    print(f"Hello {name}!")
 
-When using the Python `deploy` method specify a flow storage location other than a Docker image requires the `flow.from_source` method. The `source` and `entrypoint` arguments are required. 
 
-The `entrypoint` is the path to the file the flow is located in and the function name, separated by a colon.
+if __name__ == "__main__":
+    my_flow.from_source(
+        source=str(Path(__file__).parent),  # code stored in local directory
```

Comment:
stray thought: we should find a more ergonomic way to do this

Reply:
100%